components/user-mgt/org.wso2.carbon.identity.user.pre.update.profile.action/src/main/java/org/wso2/carbon/identity/user/pre/update/profile/action/internal/management/PreUpdateProfileActionDTOModelResolver.java

```diff
@@ -207,4 +226,43 @@ private List<String> validateAttributes(Object attributes) throws ActionDTOModel
 
         return (List<String>) attributes;
     }
+
+    private void validateAttributesCount(List<String> attributes) throws ActionDTOModelResolverClientException {
+
+        if (attributes.size() > 10) {
+            throw new ActionDTOModelResolverClientException("Invalid number of attributes.",
+                    "Maximum number of attributes allowed is 10.");
```

Comment:
Let's update the message something similar to
"Maximum number of allowed attributes to configure exceeded. Maximum allowed: {max number} Provider: {current list size}